### PR TITLE
feat: removed descriptions and added info.title as group names for x-tagGroup  in Join command

### PR DESCRIPTION
**Pull request by o-bbk**
## What/Why/How?
Removed descriptions adding for x-tagGroups for the `join` command as they are not supported and cause errors on linting.

## Reference
Resolves https://github.com/Redocly/redocly-cli/issues/1203
Resolves https://github.com/Redocly/redocly-cli/issues/1314

## Check yourself

- [x] Code is linted
- [ ] Tested with redoc/reference-docs/workflows (internal)
- [x] All new/updated code is covered with tests

## Security

- [x] Security impact of change has been considered
- [x] Code follows company security practices and guidelines

### CODE REVIEW

File: .changeset/grumpy-toys-reply.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@redocly/cli": patch
+---
+
+Removed descriptions adding for x-tagGroups for the `join` command.
```
Comment (lornajane) [inline .changeset/grumpy-toys-reply.md:5]: Please include something about why this change was made, and the differences that the user may observe.

**Review by lornajane — DISMISSED** [review]

**Review by tatomyr — APPROVED** [review]

**Comment by lornajane** [comment]
Feature does work perfectly though - just the changelog needs some work :)

File: .changeset/grumpy-toys-reply.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+"@redocly/cli": patch
+---
+
+- Removed descriptions adding for x-tagGroups for the `join` command. Descriptions in x-tagGroups are not supported and cause errors on linting.
+- `info.title` is now used as a name in x-tagGroups instead of a file name for the `join` command to make possible the joining of files with the same names.
```
Comment (lornajane) [inline .changeset/grumpy-toys-reply.md:6]: Will existing users see changed behaviour? Is it configurable?

Comment (adamaltman) [reply .changeset/grumpy-toys-reply.md:6]: What if the `info.title`'s are the same?

Comment (tatomyr) [reply .changeset/grumpy-toys-reply.md:6]: Use info versions as well? Is it a valid case to join different versions of the same API descriptions? If even that fails, I'd expect a warning/error. Alternatively, we can add indexes (like Title_1, Title_2 and so on), but this behaviour might be confusing.

Comment (lornajane) [reply .changeset/grumpy-toys-reply.md:6]: If the titles are the same, they'll only get one `x-tagGroups` entry. Let's link to the [info-override decorator](https://redocly.com/docs/cli/decorators/info-override/) in the join command docs so that the user can adjust one or all of the titles to suit their needs before joining.

Comment (malis42) [reply .changeset/grumpy-toys-reply.md:6]: This docs change will be resolved in a separate PR here https://github.com/Redocly/redocly-cli/pull/1422

**Review by tatomyr — APPROVED** [review]

File: .changeset/grumpy-toys-reply.md (hunk 2 above)
Comment (HCloward) [inline .changeset/grumpy-toys-reply.md:6]:
```suggestion
- Updated `info.title` to be used as a name in x-tagGroups instead of a file name for the `join` command, so you can now join files with the same names.
```

Comment (malis42) [reply .changeset/grumpy-toys-reply.md]: Fixed, thanks!

**Review by HCloward — APPROVED** [review]

File: .changeset/grumpy-toys-reply.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+---
+"@redocly/cli": patch
```
[the remark this answers is not in the document]
Comment (malis42) [reply .changeset/grumpy-toys-reply.md]: Changed to minor as requested, thanks!
